graphstorm-processing/tests/test_row_count_utils.py

```diff
@@ -0,0 +1,294 @@
+"""
+
+import os
+import json
+import shutil
+import pytest
+
+import pyarrow as pa
+import pyarrow.parquet as pq
+
+from graphstorm_processing.graph_loaders.row_count_utils import (
+    ParquetRowCounter,
+    verify_metadata_match,
+)
+from graphstorm_processing.constants import FilesystemType
+
+# pylint: disable=redefined-outer-name
+
+_ROOT = os.path.abspath(os.path.dirname(__file__))
+TEMP_DATA_PREFIX = os.path.join(_ROOT, "resources/row_counting/generated_parquet/")
+
+
+def create_feature_table(col_name: str, num_rows: int, feature_start_val: int) -> pa.Table:
+    """Creates a test PyArrow table."""
+    feature: pa.Array = pa.array(range(feature_start_val, feature_start_val + num_rows))
+    return pa.table([feature], names=[col_name])
+
+
+@pytest.fixture(scope="module")
+def test_metadata():
+    """Create and return the test metadata structure."""
+    return {
+        "node_type": ["type1", "type2"],
+        "edge_type": ["type1:edge:type2"],
+        "node_data": {
+            "type1": {
+                "feature1": {
+                    "format": {"name": "parquet"},
+                    "data": [
+                        "node_data/type1-feature1/part-00000.parquet",
+                        "node_data/type1-feature1/part-00001.parquet",
+                    ],
+                },
+                "feature2": {
+                    "format": {"name": "parquet"},
+                    "data": [
+                        "node_data/type1-feature2/part-00000.parquet",
+                        "node_data/type1-feature2/part-00001.parquet",
+                    ],
+                },
+            },
+            "type2": {
+                "feature1": {
+                    "format": {"name": "parquet"},
+                    "data": [
+                        "node_data/type2-feature1/part-00000.parquet",
+                        "node_data/type2-feature1/part-00001.parquet",
+                    ],
+                }
+            },
+        },
+        "edge_data": {
+            "type1:edge:type2": {
+                "weight": {
+                    "format": {"name": "parquet"},
+                    "data": [
+                        "edge_data/type1_edge_type2-weight/part-00000.parquet",
+                        "edge_data/type1_edge_type2-weight/part-00001.parquet",
+                    ],
+                }
+            }
+        },
+        "raw_id_mappings": {
+            "type1": {
+                "format": {"name": "parquet"},
+                "data": [
+                    "raw_id_mappings/type1/part-00000.parquet",
+                    "raw_id_mappings/type1/part-00001.parquet",
+                ],
+            },
+            "type2": {
+                "format": {"name": "parquet"},
+                "data": [
+                    "raw_id_mappings/type2/part-00000.parquet",
+                    "raw_id_mappings/type2/part-00001.parquet",
+                ],
+            },
+        },
+        "edges": {
```

Comment:
I see we do not test for the edge data, though it is unlikely, shall we also try to test it?

Reply:
Yeah good point, I'll add a test for edge_data as well.